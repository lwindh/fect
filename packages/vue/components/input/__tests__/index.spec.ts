import { mount } from '@vue/test-utils'
import Input from '..'

describe('Input', () => {
  it('should be render as a element', () => {
    const wrapper = mount(Input, {
      props: {
        modelValue: '',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should be support different sizes', () => {
    const wrapper = mount({
      template: `
      <div>
        <fe-input size="mini" placeholder="最小的" />
        <fe-input size="small" placeholder="较小的" />
        <fe-input size="medium" placeholder="中等的" />
        <fe-input size="large" placeholder="较大的" />
      </div>
      `,
      components: {
        [Input.name]: Input,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should be work with label', () => {
    const wrapper = mount({
      template: `
      <div>
        <fe-input size="mini" placeholder="最小的">
         <span>Test</span>
        </fe-input>
      </div>
      `,
      components: {
        [Input.name]: Input,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('label').html()).toContain('<label><span>Test</span></label>')
  })
  it('should be support prefix and suffix label', () => {
    const wrapper = mount({
      template: `
      <div>
        <fe-input  placeholder="最小的"  suffix="Suffix" prefix="Prefix"  />
      </div>
      `,
      components: {
        [Input.name]: Input,
      },
    })
    expect(wrapper.findAll('.fect-input__label')[1].classes('suffix')).toBe(true)
  })
  it('should set input from value', () => {
    const wrapper = mount(Input, {
      props: {
        type: 'text',
        modelValue: '',
      },
    })
    const input = wrapper.find('input')
    input.element.value = '1'
    input.trigger('input')
    expect(wrapper.emitted('update:modelValue')![0]).toEqual(['1'])
  })

  it('should support clearable and password icon', () => {
    const wrapper = mount(Input, {
      props: {
        clearable: true,
        type: 'password',
      },
    })
    expect(wrapper.find('.fect-input__clear-icon')).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('when prpos type is number emit value should be number type', () => {
    const wrapper = mount(Input, {
      props: {
        modelValue: 0,
        type: 'number',
      },
    })
    const input = wrapper.find('input')
    input.element.value = 2 as any
    input.trigger('input')
    expect(wrapper.emitted('update:modelValue')![0]).toEqual([2])
  })
  it('when type as passowrd when trigger icon should be text', async () => {
    const wrapper = mount(Input, {
      props: {
        modelValue: '123456',
        type: 'password',
      },
    })

    const el = wrapper.find('.fect-input__icon')
    await el.trigger('click')
    expect(wrapper.find('input').attributes('type')).toBe('text')
  })
  it('input can not work on type as disabled or readonly', () => {
    const wrapper = mount({
      template: `
      <div>
        <fe-input size="mini" placeholder="最小的" disabled />
        <fe-input size="mini" placeholder="最小的" readonly />
      </div>
      `,
      components: {
        [Input.name]: Input,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('when input focus should change border color', async () => {
    const wrapper = mount(Input)
    const el = wrapper.find('input')
    await el.trigger('focus')
    expect(wrapper.find('.hover')).toBeTruthy()
    await el.trigger('blur')
    expect(wrapper.find('.hover').exists()).toBe(false)
  })
  it('clearHandler should be work', async () => {
    const wrapper = mount(Input, {
      props: {
        modelValue: 333,
        clearable: true,
      },
    })
    const el = wrapper.find('.fect-input__clear-icon')
    expect(wrapper.find('.visible').exists()).toBe(true)
    await wrapper.setProps({ disabled: true })
    await wrapper.find('input').trigger('change')
    expect(wrapper.find('.disabled').exists()).toBe(true)
    await wrapper.setProps({ disabled: false })
    await wrapper.find('input').trigger('change')
    await el.trigger('click')
    const emited = wrapper.emitted('update:modelValue')![0] as any
    expect(emited[0]).toBe('')
  })
})
