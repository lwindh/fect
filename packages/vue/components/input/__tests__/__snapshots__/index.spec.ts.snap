// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input input can not work on type as disabled or readonly 1`] = `
<div>
  <div class="fect-input" style="--heightRatio: 1.313;">
    <!---->
    <div class="fect-input__container">
      <!---->
      <div class="fect-input__wrapper  disabled"><input type="text" class="disabled" style="font-size: 12px;" disabled="" placeholder="最小的">
        <!---->
        <!---->
      </div>
      <!---->
    </div>
  </div>
  <div class="fect-input" style="--heightRatio: 1.313;">
    <!---->
    <div class="fect-input__container">
      <!---->
      <div class="fect-input__wrapper  "><input type="text" class="" style="font-size: 12px;" readonly="" placeholder="最小的">
        <!---->
        <!---->
      </div>
      <!---->
    </div>
  </div>
</div>
`;

exports[`Input should be render as a element 1`] = `
<div class="fect-input" style="--heightRatio: 1.687;">
  <!---->
  <div class="fect-input__container">
    <!---->
    <div class="fect-input__wrapper  "><input type="text" class="" style="font-size: 14px;">
      <!---->
      <!---->
    </div>
    <!---->
  </div>
</div>
`;

exports[`Input should be support different sizes 1`] = `
<div>
  <div class="fect-input" style="--heightRatio: 1.313;">
    <!---->
    <div class="fect-input__container">
      <!---->
      <div class="fect-input__wrapper  "><input type="text" class="" style="font-size: 12px;" placeholder="最小的">
        <!---->
        <!---->
      </div>
      <!---->
    </div>
  </div>
  <div class="fect-input" style="--heightRatio: 1.5;">
    <!---->
    <div class="fect-input__container">
      <!---->
      <div class="fect-input__wrapper  "><input type="text" class="" style="font-size: 12px;" placeholder="较小的">
        <!---->
        <!---->
      </div>
      <!---->
    </div>
  </div>
  <div class="fect-input" style="--heightRatio: 1.687;">
    <!---->
    <div class="fect-input__container">
      <!---->
      <div class="fect-input__wrapper  "><input type="text" class="" style="font-size: 14px;" placeholder="中等的">
        <!---->
        <!---->
      </div>
      <!---->
    </div>
  </div>
  <div class="fect-input" style="--heightRatio: 1.875;">
    <!---->
    <div class="fect-input__container">
      <!---->
      <div class="fect-input__wrapper  "><input type="text" class="" style="font-size: 16px;" placeholder="较大的">
        <!---->
        <!---->
      </div>
      <!---->
    </div>
  </div>
</div>
`;

exports[`Input should be work with label 1`] = `
<div>
  <div class="fect-input" style="--heightRatio: 1.313;"><label><span>Test</span></label>
    <div class="fect-input__container">
      <!---->
      <div class="fect-input__wrapper  "><input type="text" class="" style="font-size: 12px;" placeholder="最小的">
        <!---->
        <!---->
      </div>
      <!---->
    </div>
  </div>
</div>
`;

exports[`Input should support clearable and password icon 1`] = `
<div class="fect-input" style="--heightRatio: 1.687;">
  <!---->
  <div class="fect-input__container">
    <!---->
    <div class="fect-input__wrapper  "><input type="password" class="" style="font-size: 14px;">
      <div class="fect-input__clear-icon "><svg viewBox="0 0 24 24" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision">
          <path d="M18 6L6 18"></path>
          <path d="M6 6l12 12"></path>
        </svg></div>
      <div class="fect-input__icon" style="cursor: pointer; pointer-events: auto;"><svg viewBox="0 0 24 24" width="16" height="16" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision" style="color: currentColor;">
          <path d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"></path>
          <circle cx="12" cy="12" r="3"></circle>
        </svg></div>
    </div>
    <!---->
  </div>
</div>
`;
